packages/limit-service/lib/limit.js

```diff
@@ -98,8 +98,13 @@ class MaxLimit extends Limit {
         return new this.errors.HostLimitError(errorObj);
     }
 
-    async currentCountQuery() {
-        return await this.currentCountQueryFn(this.db);
+    /**
+     * @param {Object} [options]
+     * @param {Object} [options.transacting] Transaction to run the count query on
+     * @returns 
+     */
+    async currentCountQuery(options = {}) {
+        return await this.currentCountQueryFn(options.transacting ? options.transacting : (this.db ? this.db.knex : undefined));
```

Comment:
This makes total sense and is a correct approach imo! 
One small nit - double nested ternary operator is soooooooooooo hard to read. It never passes the "single glance" test. Sometimes just having couple nested if-elses is way more readable, like in this situation :upside_down_face: 

Maybe we could make use the optional chaining with nullish coalescing here to make it like `options.transacting? ?? this.db?.knex?`... although after stearing at it might be pretty hard to reason too :smile:

Reply:
I've bumped the `ecmaVersion` to 2020 in the eslint file to make it work.